core/templates/dev/head/components/promo/PromoBarDirective.js

```diff
@@ -19,25 +19,33 @@
  */
 
 oppia.directive('promoBar', [
-  'UrlInterpolationService', function(UrlInterpolationService) {
+  'PromoBarService', 'UrlInterpolationService',
+  function(PromoBarService, UrlInterpolationService) {
     return {
       restrict: 'E',
-      scope: {
-        getPromoMessage: '&promoMessage'
-      },
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/components/promo/' +
         'promo_bar_directive.html'),
       controller: [
-        '$scope',
-        function($scope) {
+        '$scope', '$window',
+        function($scope, $window) {
           var isPromoDismissed = function() {
             return !!angular.fromJson(sessionStorage.promoIsDismissed);
           };
           var setPromoDismissed = function(promoIsDismissed) {
             sessionStorage.promoIsDismissed = angular.toJson(promoIsDismissed);
           };
 
+          if ($window.location.pathname.split('/')[1] === 'signup') {
```

Comment:
Why is this `if` needed?

Reply:
I'm not sure that's the right fix. We should probably update the backend handler so that it can be pinged anytime -- whether the promo bar appears or not shouldn't depend on whether the user is logged in or logged out.

Reply:
Whether promo bar appears or not does not depend on whether user is logged in or logged out. Due to `if` promo bar will not appear in user registration page. Following is the user registration page:
![user](https://user-images.githubusercontent.com/15020747/52290265-b2508c80-2995-11e9-80cc-1a73ec2d85d4.png)

Reply:
What I was trying to say was that the promo bar _should_ appear in the user registration page -- I don't see a reason not to show it there.

If there is a "session expired" error in that case, we should find out why that is happening and fix it, rather than introduce a special case here.

Reply:
I discovered that the "session expired" error was occurring because partially logged in user was being logged out in `base.py`. To prevent this in case of promo bar, variable `LOGOUT_PARTIALLY_LOGGED_IN_USER` was introduced in `class BaseHandler`.

Reply:
No, this is messy (at first glance). I don't think it's a good long-term solution; introducing new special cases often isn't.

Could you please explain the action flow during logout using a [sequence diagram](https://en.wikipedia.org/wiki/Sequence_diagram)? This is similar to what I asked @ankita240796 in https://github.com/oppia/oppia/pull/6170#issuecomment-458860194. With that tool, we were able to analyze the flow and figure out a good solution.

In general it's really important to understand very clearly what exactly is causing the current behaviour. That would allow us to find the best fix for it.

Reply:
Also, it might be worth to cooperate with @YashJipkate since as he mentioned he is having a similar issue.

Reply:
Hi @seanlip . [Here](https://docs.google.com/document/d/1dUeco3Z--a7lktYv8Xn4YURNSG1ZZkAdLsc_tlDVh7g/edit?usp=sharing) is the sequence diagram. During the request made to `promo_bar_handler`, user is partially logged in and hence is logged out as can be seen [here](https://github.com/oppia/oppia/blob/develop/core/controllers/base.py#L227). One solution I found out to solve this problem is to add `REDIRECT_UNFINISHED_SIGNUPS = False` in `PromoBarHandler`. PTAL, Thanks!

Reply:
I think the solution you propose is appropriate.

Reply:
I agree with @vojtechjelinek.